packages/@aws-cdk/cloud-assembly-schema/lib/schema.ts

```diff
@@ -0,0 +1,90 @@
+import * as fs from 'fs';
+import * as jsonschema from 'jsonschema';
+import * as assembly from './manifest';
+
+/**
+ * Protocol utility class.
+ */
+export class Manifest {
+  /**
+   * Save manifest to file.
+   *
+   * @param manifest - manifest.
+   */
+  public static save(manifest: assembly.AssemblyManifest, filePath: string) {
+    fs.writeFileSync(filePath, JSON.stringify(manifest, undefined, 2));
+  }
+
+  /**
+   * Load manifest from file.
+   */
+  public static load(filePath: string): assembly.AssemblyManifest {
+    const raw: assembly.AssemblyManifest = JSON.parse(fs.readFileSync(filePath, 'UTF-8'));
+    Manifest.patchStackTags(raw);
+    Manifest.validate(raw);
+    return raw;
+  }
+
+  /**
+   * Fetch the current schema version number.
+   */
+  public static version(): string {
+    // eslint-disable-next-line @typescript-eslint/no-require-imports
+    return require('../schema/cloud-assembly.metadata.json').version;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-require-imports
+  private static schema: jsonschema.Schema = require('../schema/cloud-assembly.schema.json');
+
+  private static validate(manifest: assembly.AssemblyManifest) {
```

Comment:
I expected version check to happen here. Should be straightforward: this should fail if you are trying to load a cloud assembly with a newer schema version.

Reply:
Yeap, i'll move it.